app/api/sul_bib/publications_api.rb

```diff
@@ -10,13 +10,11 @@ class PublicationsAPI < Grape::API
       #         returns an error if the publication is not found or if it has
       #         been deleted (unless the request is a 'DELETE')
       def publication_find(id)
-        begin
-          pub = Publication.find(id)
-        rescue ActiveRecord::RecordNotFound
-          error!({ 'error' => 'No such publication', 'detail' => "You've requested a non-existant publication." }, 404)
-        end
+        pub = Publication.find(id)
         error!('Gone - old resource deleted.', 410) if pub.deleted? && !request.delete?
         pub
+      rescue ActiveRecord::RecordNotFound
+        error!({ 'error' => 'No such publication', 'detail' => "You've requested a non-existant publication." }, 404)
       end
```

Comment:
Generally :+1: on this change
The extra check on the delete here adds a responsibility to this method that wasn't there before; code climate is complaining about the extra responsibility.  I get that the single responsibility principle might be a concern, but I'm not entirely clear on the whole context of how this method is used in this part of the API.  If it helps to DRY the code, then maybe this is best place for it.  I don't have any strong preferences about it, just commenting.
It's curious that the `error!` is accepting either a `String` or a `Hash` here.  I'm not entirely familiar with where this error method is defined (rails, grape, ?); just wanting to confirm that it does accept a `String` and also that when it is called by a `Hash` whether that has some impact on how the CAP-team client can parse the error response (if they do).